src/associate-current-user.js

```diff
@@ -17,16 +20,16 @@ export default function (options = {}) {
       throw new Error('There is no current user to associate.');
     }
 
-    options = Object.assign({}, defaults, hook.app.get('auth'), options);
+    options = Object.assign({}, defaults, hook.app.get('authentication'), options);
 
-    const id = hook.params.user[options.idField];
+    const id = get(hook.params.user, options.idField);
```

Comment:
It probably won't be but it allows to set a dot-separated `idField` like `user.data.id`. I just changed this for all fields.